### ISSUE TITLE
URLRouter doesn't build paths with root_path from scope

### ISSUE THREAD

**Issue by ajslater** [django/channels#1973]
It seems to me that here: https://github.com/django/channels/blob/main/channels/routing.py#L100

URLRouter should strip the `scope['root_path']` from the submitted path for matching against paths.

Django does this for path matching.

### Workaround

Manually add the configured asgi root_path to the head of all paths submitted to the UrlRouter().

e.g. 
```python
PREFIX = HYPERCORN_CONFIG.root_path[1:] + "/" if HYPERCORN_CONFIG.root_path else ""

URLRouter([
    path(f"{PREFIX}ws", MyConsumer.as_asgi()),  
])
```

**Comment by carltongibson** [comment]
Fixed in #1954

**Comment by ajslater** [comment]
This is fixed. Thank you!

In related news, the Django router also stopped setting script_prefix from scope root_path two months ago. This is not Channels' problem, but might be relevant to anyone searching on this bug.
A workaround is attached to this bug report: https://code.djangoproject.com/ticket/35169#comment:11
